projects/igniteui-angular/src/lib/calendar/calendar-base.ts

```diff
@@ -641,20 +695,55 @@ export class IgxCalendarBaseDirective implements ControlValueAccessor {
 
             this.selectedDates = Array.from(new Set([...newDates, ...selDates])).map(v => new Date(v));
         } else {
-            const valueDateOnly = this.getDateOnly(value);
-            const newSelection = [];
-            if (this.selectedDates.every((date: Date) => date.getTime() !== valueDateOnly.getTime())) {
-                newSelection.push(valueDateOnly);
+            let newSelection = [];
+
+            if (this.shiftKey && this.lastSelectedDate) {
+
+                [this._startDate, this._endDate] = this.lastSelectedDate.getTime() < value.getTime()
+                    ? [this.lastSelectedDate, value]
+                    : [value, this.lastSelectedDate];
+
+                const unselectedDates = [this._startDate, ...this.generateDateRange(this._startDate, this._endDate)]
+                    .filter(date => !this.isDateDisabled(date)
+                        && this.selectedDates.every((d: Date) => d.getTime() !== date.getTime())
```

Comment:
```
this.selectedDates.every((d: Date) => d.getTime() !== date.getTime()
```
This goes through all selected dates and check if any date differs from date.

Seems to be possible to use `some` operator and check if any of the selected dates is the same as date and use the logical not operator:

```
!this.selectedDates.some((d: Date) => d.getTime() === date.getTime()
```

It is not drama anyway, wont kill performance